atlite/cutout.py

```diff
@@ -42,16 +43,21 @@
 class Cutout(object):
     dataset_module = None
 
-    def __init__(self, name=None, data=None, cutout_dir=".", **cutoutparams):
+    def __init__(self, name=None, data=None, cutout_dir=None, **cutoutparams):
+
         if isinstance(name, xr.Dataset):
             data = name
             name = data.attrs.get("name", "unnamed")
 
-        if '/' in name:
-            cutout_dir, name = os.path.split(name)
+        if name:
+            cutout_dir = os.path.dirname(name)
+            name = os.path.basename(name)
+
+            self.cutout_dir = cutout_dir
+            self.name = name
```

Comment:
The relevant line is not visible in the four lines shown by github above, you will have to open the [diff](https://github.com/FRESNA/atlite/pull/23/files/092ddf7be6b074b647606d815c85a923199a038c#diff-58ca2325f6111c72edbaa911edcc20be) directly for the correct context.

Reply:
You're right, it does not make sense.
The idea was to overwrite the cutout dir if there is a valid directory given in the `name`.
But that possibility is given by the `cutout_dir` function parameter anyways.

Reply:
Well, the idea that one can just provide a full path as first argument is a powerful one. It is exactly what the current code
https://github.com/FRESNA/atlite/blob/c321dae59e083ecdba88b283165c782f738f405d/atlite/cutout.py#L48-L49
achieves, isn't it?

Reply:
That is what intends to do but is not portable to windows.
Thanks for pointing out the unintentional overwrite, I tried to fix it, see if you think the new (but terrible looking) solution works for you.

Otherwise for a nice looking and robust code, we could just do a
```
os.path.basename(name)
```

and force the user to specify a `cutout_dir` with the intended parameter.